### PR TITLE
fix(BTabs): fix recursion on titleLinkAttrs

### LINKED ISSUE #2167: title-link-attrs in BTab leads to infinite recursive updates
### Describe the bug

When adding title-link-attrs to BTab, we get an infinite update recursion.

```
Uncaught (in promise) Maximum recursive updates exceeded. This means you have a reactive effect that is mutating its own dependencies and thus recursively triggering itself. Possible sources include component template, render function, updated hook or watcher source function.
```

### Reproduction

https://github.com/dword-design-honestly/demo-bootstrap-vue-next-title-link-attrs

### Used Package Manager

pnpm

### Validations

- [X] Have tested with the latest version. This is still alpha version and sometime things change rapidly.
- [X] Follow our [Code of Conduct](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by xvaara**
# Describe the PR

closes #2167 

## PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix :bug: - `fix(...)`
- [ ] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [ ] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [x] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

### CODE REVIEW

File: packages/bootstrap-vue-next/src/components/BTabs/BTab.vue [hunk 1]
```diff
@@ -67,7 +67,7 @@ const tab = computed(
       title: props.title,
       titleComponent: slots.title,
       titleItemClass: props.titleItemClass,
-      titleLinkAttrs: props.titleLinkAttrs,
+      titleLinkAttrs: toRef(() => props.titleLinkAttrs),
```
Comment (xvaara) [inline packages/bootstrap-vue-next/src/components/BTabs/BTab.vue:70]: tested, this works if this line is just `titleLinkAttrs: () => props.titleLinkAttrs` and v-bind is just `v-bind="tab.titleLinkAttrs?.()"`

Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BTabs/BTab.vue:70]: @VividLemon which would be better? it seems v-bind with and object causes all the childs props to update which updates v-bind object and loops.

**Comment by xvaara** [comment]
> Note that this is also the case for other props like titleLinkClass if an object is passed.

Nice catch! I'll fix them too.

**Comment by dword-design-honestly** [comment]
Let me know if I can help here 🙂.

File: packages/bootstrap-vue-next/src/components/BTabs/BTab.vue (hunk 1 above)
Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BTabs/BTab.vue:70]:
```
<b-tabs>
    <b-tab title="Tab 1" :title-link-attrs="{ foo: 'bar'}" />
  </b-tabs>
 ```
does not work. And I think that is what most people will use.

Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BTabs/BTab.vue:70]: Maximum recursive updates exceeded in component <BTabs>. This means you have a reactive effect that is mutating its own dependencies and thus recursively triggering itself. Possible sources include component template, render function, updated hook or watcher source function.

https://stackblitz.com/github/dword-design-honestly/demo-bootstrap-vue-next-title-link-attrs

**Review by VividLemon — APPROVED** [review]
This doesn't seem like the best way to fix this issue. But seems fine

**Pull request merged**